### DISCUSSION TITLE
Wrong info from TS0601_air_quality_sensor

### DISCUSSION THREAD

**Discussion by mak-42** [Koenkk/zigbee2mqtt#8945] [Devices]
I've connected TS0601_air_quality_sensor (Tuya Air quality sensor), but it reports strange values. It seems "formaldehyd" value should be "co2". "voc" and "wrong co2" values are very low.

Have you got any ideas how to fix it, or I have to change the device?

> info  2021-10-01 19:05:22: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":1,"formaldehyd":358,"humidity":44,"linkquality":78,"temperature":24.4,"voc":0}'
info  2021-10-01 19:05:22: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":1,"formaldehyd":358,"humidity":44,"linkquality":78,"temperature":24.4,"voc":0}'
info  2021-10-01 19:05:22: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":1,"formaldehyd":358,"humidity":44,"linkquality":78,"temperature":24.4,"voc":0}'
info  2021-10-01 19:05:22: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":1,"formaldehyd":358,"humidity":44,"linkquality":78,"temperature":24.4,"voc":1}'
info  2021-10-01 19:05:22: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":2,"formaldehyd":358,"humidity":44,"linkquality":78,"temperature":24.4,"voc":1}'
info  2021-10-01 19:05:22: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":2,"formaldehyd":358,"humidity":44,"linkquality":78,"temperature":24.4,"voc":0}'
info  2021-10-01 19:05:22: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":1,"formaldehyd":358,"humidity":44,"linkquality":78,"temperature":24.4,"voc":0}'
info  2021-10-01 19:05:22: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":1,"formaldehyd":358,"humidity":44,"linkquality":78,"temperature":24.4,"voc":0}'
info  2021-10-01 19:05:22: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":1,"formaldehyd":358,"humidity":44,"linkquality":81,"temperature":24.4,"voc":1}'
info  2021-10-01 19:05:22: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":2,"formaldehyd":358,"humidity":44,"linkquality":81,"temperature":24.4,"voc":1}'
info  2021-10-01 19:05:22: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":2,"formaldehyd":358,"humidity":44,"linkquality":78,"temperature":24.4,"voc":0}'
info  2021-10-01 19:05:22: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":1,"formaldehyd":358,"humidity":44,"linkquality":78,"temperature":24.4,"voc":0}'
info  2021-10-01 19:05:22: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":1,"formaldehyd":358,"humidity":44,"linkquality":78,"temperature":24.4,"voc":1}'
info  2021-10-01 19:05:22: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":2,"formaldehyd":358,"humidity":44,"linkquality":78,"temperature":24.4,"voc":1}'
info  2021-10-01 19:05:22: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":2,"formaldehyd":368,"humidity":44,"linkquality":78,"temperature":24.4,"voc":1}'
info  2021-10-01 19:05:23: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":2,"formaldehyd":368,"humidity":44,"linkquality":78,"temperature":24.4,"voc":1}'
info  2021-10-01 19:05:23: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":2,"formaldehyd":368,"humidity":44,"linkquality":81,"temperature":24.4,"voc":1}'
info  2021-10-01 19:05:23: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":2,"formaldehyd":368,"humidity":44,"linkquality":81,"temperature":24.4,"voc":1}'
info  2021-10-01 19:05:23: MQTT publish: topic 'zigbee2mqtt/TS0601AirQualitySensor', payload '{"co2":2,"formaldehyd":368,"humidity":44,"linkquality":78,"temperature":24.4,"voc":0}'

**Comment by 6zXo8aMn2TRXOs** [comment]
Guess it is related to this device. https://github.com/Koenkk/zigbee2mqtt/discussions/7299

**Reply by mak-42** [reply]
It seems you are right.
